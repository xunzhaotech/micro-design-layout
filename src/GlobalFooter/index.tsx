import './index.less';

import { WithFalse } from '../typings';
import { defineComponent, PropType, SetupContext, VNodeChild } from 'vue';

export type Links = WithFalse<
  {
    key?: string;
    title: VNodeChild | JSX.Element;
    href: string;
    blankTarget?: boolean;
  }[]
>;

export interface GlobalFooterProps {
  links?: Links;
  copyright?: VNodeChild | JSX.Element;
  prefixCls?: string;
}

export default defineComponent({
  name: 'GlobalFooter',
  props: {
    links: [Array, Boolean] as PropType<Links>,
    copyright: {
      type: [Object, Function, Boolean] as PropType<VNodeChild | JSX.Element>,
      default: () => undefined,
    },
    prefixCls: {
      type: String,
      default: 'ant-pro',
    },
  },
  setup(props: GlobalFooterProps, { slots }: SetupContext) {
    if (
      (props.links == null ||
        props.links === false ||
        (Array.isArray(props.links) && props.links.length === 0)) &&
      (props.copyright == null || props.copyright === false)
    ) {
      console.warn('[pro-layout]: GlobalFooter required `links` or `copyright`');
      return () => null;
    }

    const baseClassName = `${props.prefixCls}-global-footer`;
    const copyright = props.copyright || (slots.copyright && slots.copyright());

    return () => (
      <footer class={baseClassName}>
        {props.links && (
          <div class={`${baseClassName}-links`}>
            {props.links.map(link => (
              <a
                key={link.key}
                title={link.key}
                target={link.blankTarget ? '_blank' : '_self'}
                href={link.href}
              >
                {link.title}
              </a>
            ))}
          </div>
        )}
        {props.copyright && <div class={`${baseClassName}-copyright`}>{copyright}</div>}
      </footer>
    );
  },
});
