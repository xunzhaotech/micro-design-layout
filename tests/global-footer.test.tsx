import { mount } from '@vue/test-utils';
import GlobalFooter from '../src/GlobalFooter';

const testLinks = [
  {
    key: '1',
    title: 'Pro Layout',
    href: 'https://www.github.com/vueComponent/pro-layout',
    blankTarget: true,
  },
  {
    key: '2',
    title: 'Github',
    href: 'https://www.github.com/vueComponent/ant-design-vue-pro',
    blankTarget: true,
  },
  {
    key: '3',
    title: '@Sendya',
    href: 'https://www.github.com/sendya/',
    blankTarget: true,
  },
];

describe('GlobalFooter', () => {
  it('ðŸ¥© base use', () => {
    const wrapper = mount({
      render() {
        return (
          <GlobalFooter
            links={testLinks}
            copyright={
              <a href="https://github.com/vueComponent" target="_blank">
                vueComponent
              </a>
            }
          />
        );
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ðŸ˜„ custom links', () => {
    const wrapper = mount({
      render() {
        return (
          <GlobalFooter
            links={[
              {
                key: '1',
                title: 'Pro Layout',
                href: 'https://www.github.com/vueComponent/pro-layout',
                blankTarget: true,
              },
            ]}
          />
        );
      },
    });
    const links = wrapper.findAll('.ant-pro-global-footer-links a');
    expect(links).toHaveLength(1);
  });

  it('ðŸ˜„ custom copyright', () => {
    const wrapper = mount({
      render() {
        return <GlobalFooter copyright={<a href="#copyright">vueComponent</a>} />;
      },
    });
    expect(wrapper.find('.ant-pro-global-footer-copyright a').attributes()).toHaveProperty(
      'href',
      '#copyright',
    );
  });
});
