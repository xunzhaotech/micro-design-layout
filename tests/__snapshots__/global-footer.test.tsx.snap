// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFooter ðŸ¥© base use 1`] = `
<footer class="ant-pro-global-footer">
  <div class="ant-pro-global-footer-links"><a title="1" target="_blank" href="https://www.github.com/vueComponent/pro-layout">Pro Layout</a><a title="2" target="_blank" href="https://www.github.com/vueComponent/ant-design-vue-pro">Github</a><a title="3" target="_blank" href="https://www.github.com/sendya/">@Sendya</a></div>
  <div class="ant-pro-global-footer-copyright"><a href="https://github.com/vueComponent" target="_blank">vueComponent</a></div>
</footer>
`;
